src/lib/tvgPictureImpl.h

```diff
@@ -34,43 +34,72 @@ struct Picture::Impl
 {
     unique_ptr<Loader> loader = nullptr;
     Paint* paint = nullptr;
+    uint32_t *pixels = nullptr;
+    Picture *picture = nullptr;
+    void *edata = nullptr;              //engine data
+    uint32_t flag = RenderUpdateFlag::None;
 
-    bool dispose(RenderMethod& renderer)
+    Impl(Picture* p) : picture(p)
     {
-        if (!paint) return false;
+    }
 
-        paint->pImpl->dispose(renderer);
-        delete(paint);
+    bool dispose(RenderMethod& renderer)
+    {
+        if (paint) {
+            paint->pImpl->dispose(renderer);
+            delete(paint);
 
-        return true;
+            return true;
+        }
+        else if (pixels) {
+            return renderer.dispose(edata);
+        }
+        return false;
     }
 
     void reload()
```

Comment:
i think we can return RenderUpdateFlag::Image or RenderUpdateFlag::None here if reloaded actual data.
if so Picture doesn't need to have additional flag.

Reply:
changed return vlaue to uint32_t and use it